audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java

```diff
@@ -183,19 +205,23 @@ void testGoodAndBadSuccessRequestSet() {
         Source.fromJavaStream(() -> requests)
                 .runWith(Producer.plainSink(producerSettings), akkaMaterializer);
 
+        TimeUnit.SECONDS.sleep(1);
```

Comment:
If you're looking for more reliable testing runs, the `.runWith` will return the materialized value, likely a `CompletionStage`.
```suggestion
        Source.fromJavaStream(() -> requests)
                .runWith(Producer.plainSink(producerSettings), akkaMaterializer)
                .toCompletableFuture().join();

        TimeUnit.SECONDS.sleep(1);
```

Reply:
Ditto this across the test class

Reply:
Consider also parameterising how long that timed sleep is in some way

Reply:
I've updated all the tests to include the `.toCompletableFuture().join()`.